src/v2/guide/components.md

```diff
@@ -1,1481 +1,597 @@
+    ]
   }
-}
+})
 ```
 
-Now all our counters each have their own internal state:
+Then want to render a component for each one:
 
-{% raw %}
-<div id="example-2-5" class="demo">
-  <my-component></my-component>
-  <my-component></my-component>
-  <my-component></my-component>
-</div>
-<script>
-Vue.component('my-component', {
-  template: '<button v-on:click="counter += 1">{{ counter }}</button>',
-  data: function () {
-    return {
-      counter: 0
-    }
-  }
-})
-new Vue({
-  el: '#example-2-5'
-})
-</script>
-{% endraw %}
+```html
+<blog-post
+  v-for="post in posts"
+  v-bind:key="post.id"
+  v-bind:title="post.title"
+></blog-post>
+```
+
+Above, you'll see that we can use `v-bind` to dynamically pass props. This is especially useful when you don't know the exact content you're going to render ahead of time, like when [fetching posts from an API](https://jsfiddle.net/chrisvfritz/sbLgr0ad).
 
-### Composing Components
+That's all you need to know about props for now, but once you've finished reading this page and feel comfortable with its content, we recommend coming back later to read the full guide on [Props](components-props.html).
 
-Components are meant to be used together, most commonly in parent-child relationships: component A may use component B in its own template. They inevitably need to communicate to one another: the parent may need to pass data down to the child, and the child may need to inform the parent of something that happened in the child. However, it is also very important to keep the parent and the child as decoupled as possible via a clearly-defined interface. This ensures each component's code can be written and reasoned about in relative isolation, thus making them more maintainable and potentially easier to reuse.
+## A Single Root Element
 
-In Vue, the parent-child component relationship can be summarized as **props down, events up**. The parent passes data down to the child via **props**, and the child sends messages to the parent via **events**. Let's see how they work next.
+When building out a `<blog-post>` component, your template will eventually contain more than just the title:
 
-<p style="text-align: center;">
-  <img style="width: 300px;" src="/images/props-events.png" alt="props down, events up">
-</p>
+```html
+<h3>{{ post.title }}</h3>
+```
 
-## Props
+At the very least, you'll want to include the post's content:
 
-### Passing Data with Props
+```html
+<h3>{{ post.title }}</h3>
+<div v-html="post.content"></div>
+```
 
-Every component instance has its own **isolated scope**. This means you cannot (and should not) directly reference parent data in a child component's template. Data can be passed down to child components using **props**.
+If you try this in your template however, Vue will show an error, explaining that **every component must have a single root element**. Vue enforces this limitation to enable some interesting features, which you'll learn about later.
```

Comment:
This is an actual technical limitation during the implementation of Vue 2, and we may eventually support fragments as component template root in the future, so I'd avoid phrasing it as if we did it to enable specific features.

Reply:
Fixed. 👍